llama-index-core/llama_index/core/base/llms/types.py

```diff
@@ -82,8 +62,49 @@ class ChatMessage(BaseModel):
     """Chat message."""
 
     role: MessageRole = MessageRole.USER
-    content: Optional[Any] = ""
-    additional_kwargs: dict = Field(default_factory=dict)
+    additional_kwargs: dict[str, Any] = Field(default_factory=dict)
+    blocks: list[ContentBlock] = Field(default_factory=list)
+
+    def __init__(self, /, content: Any | None = None, **data: Any) -> None:
+        """Keeps backward compatibility with the old `content` field.
+
+        If content was passed and contained text, store a single TextBlock.
+        If content was passed and it was a list, assume it's a list of content blocks and store it.
+        """
+        if content is not None:
+            if isinstance(content, str):
+                data["blocks"] = [TextBlock(text=content)]
+            elif isinstance(content, list):
+                data["blocks"] = content
+
+        super().__init__(**data)
+
+    @property
+    def content(self) -> str:
```

Comment:
slightly scared to move `content` to a python property vs. field

I wonder what this does to stuff like the `RedisChatStore` (or any chat message that was previously serialized)

Reply:
I'll double check and add tests. This is supposed to be compatible with existing code but it's true that a full roundtrip disk -> memory -> disk of an existing serialized object would result in a different format on disk. I couldn't think of a use case where this would be a problem though.